gapic/src/main/com/google/gapid/views/ActivityPickerDialog.java

```diff
@@ -171,67 +263,111 @@ protected void showError(String message) {
-        TreeItem activityItem = new TreeItem(pkgItem, 0);
-        activityItem.setText(activity.getName());
-
-        for (PkgInfo.Action action : activity.getActionsList()) {
-          TreeItem actionItem = new TreeItem(activityItem, 0);
-          actionItem.setText(action.getName());
-          if (action.getIsLaunch() && launchAction == null) {
-            actionItem.setFont(boldFont);
-            launchAction = new Action(pkg, activity, action);
-          }
-          actionItem.setData(new Action(pkg, activity, action));
-        }
-        if (launchAction != null && launchAction.activity == activity) {
-          activityItem.setFont(boldFont);
-          activityItem.setData(launchAction);
-        }
-        int activityIconIdx = activity.getIcon();
-        if (activityIconIdx < 0 || activityIconIdx >= iconCount) {
-          activityIconIdx = pkgIconIdx;
-        }
-        activityItem.setImage((activityIconIdx >= 0 && activityIconIdx < iconCount) ?
-            icons[activityIconIdx] : noIcon);
+    @Override
+    public boolean hasChildren(Object element) {
+      if (element instanceof PkgInfo.PackageList) {
+        return ((PkgInfo.PackageList)element).getPackagesCount() > 0;
+      } else if (element instanceof PkgInfo.Package) {
+        return ((PkgInfo.Package)element).getActivitiesCount() > 0;
+      } else if (element instanceof PkgInfo.Activity) {
+        return ((PkgInfo.Activity)element).getActionsCount() > 0;
+      } else {
+        return false;
       }
-      if (launchAction != null) {
-        pkgItem.setData(launchAction);
+    }
+
+    @Override
+    public Object[] getChildren(Object element) {
+      if (element instanceof PkgInfo.PackageList) {
+        return ((PkgInfo.PackageList)element).getPackagesList().toArray();
+      } else if (element instanceof PkgInfo.Package) {
+        return ((PkgInfo.Package)element).getActivitiesList().toArray();
+      } else if (element instanceof PkgInfo.Activity) {
+        return ((PkgInfo.Activity)element).getActionsList().toArray();
+      } else {
+        return new Object[0];
+      }
+    }
+
+    @Override
+    public Object getParent(Object element) {
+      return null;
+    }
+  }
+
+  private static class PackageLabelProvider extends LabelProvider implements IStyledLabelProvider {
+    private final Theme theme;
+    private Image[] icons = new Image[0];
+
+    public PackageLabelProvider(Theme theme) {
+      this.theme = theme;
+    }
+
+    public void setIcons(Image[] icons) {
+      this.icons = icons;
+    }
+
+    @Override
+    public StyledString getStyledText(Object element) {
+      if (element instanceof PkgInfo.Package) {
+        return new StyledString(((PkgInfo.Package)element).getName());
+      } else if (element instanceof PkgInfo.Activity) {
+        PkgInfo.Activity a = (PkgInfo.Activity)element;
+        return new StyledString(
+            a.getName(), Action.findLaunchAction(a) != null ? theme.labelStyler() : null);
```

Comment:
It styles it as a label, which does mean bold. The name of the styler is not the best, it probably should be called em(phasized)Styler.